### ISSUE TITLE
[topsimulator] IceTop detector simulation: I3CorsikaInjector and .bz2 files (Trac #1788)

### ISSUE THREAD

**Issue by icecube-trac** [icecube-trac/tix2#1780]
Migrated from https://code.icecube.wisc.edu/ticket/1788
```json
{
    "status": "accepted", 
    "changetime": "2020-06-24T12:31:42", 
    "description": "Hi everyone, \ni have a technical question concerning the read-in of compressed CORSIKA files (.bz2) for the IceTop detector simulation. I use the simulation sets 10410 and 10889. \nI learned that the current version of the I3CorsikaInjector can not treat bz2 files.\nThus i build the following solution:\n\n1) Copy the compressed CORSIKA file .bz2 file to my directory\n2) Unpack it\n3) Use it\n4) Delete it\n\nThis solution works fine but it is very, very time consuming. Copying & unpacking bz2 files limits the number of jobs i can run in parallel to a very small number because i easily exceed 1TB (i already have some hundreds of GB of data which i really need). I want to run more jobs in parallel, because otherwise i still need months to finish my detector simulation. I do not have this time :(\nThe detector simulation i do bases on the detector response of different particle types (separately!) and anyway takes more time than a \"usual\" simulation.\n\nIs there a solution? Are there some lines of code i can add in order to make I3CorsikaInjector treat .bz2 files? I heard that it may be possible when \"if stream\" is changed to \"boost::iostreams::filtering_istream\", but i am completely unfamiliar with this.\nCan you please help me? \n\nCheers, Daniel", 
    "reporter": "dbindig", 
    "cc": "", 
    "resolution": "", 
    "_ts": "1593001902142004", 
    "component": "combo simulation", 
    "summary": "[topsimulator] IceTop detector simulation: I3CorsikaInjector and .bz2 files", 
    "priority": "minor", 
    "keywords": "", 
    "time": "2016-07-25T14:44:40", 
    "milestone": "Autumnal Equinox 2020", 
    "owner": "mplum", 
    "type": "enhancement"
}
```

**Comment by icecube-trac** [comment]
Trac update at 20160727T15:36:42: 
* olivas changed owner from "" to "jgonzalez"
* olivas changed status from "new" to "assigned"

**Comment by icecube-trac** [comment]
Trac update at 20160728T18:17:01: jgonzalez commented:

> In my opinion, this should not be fixed. I would resolve this ticket as "won't fix" but I want the opinion of other developers.
> 
> There are other features that are missing from the CORSIKA reader in topsimulator, but the proper thing to do is to have a single CORSIKA reader that can be used by all, and that includes all the missing features such as this, instead of having multiple corsika readers around. I have proposed such a reader last year, but so far the response has not been great.

**Comment by icecube-trac** [comment]
Trac update at 20160728T19:59:21: 
* jgonzalez changed milestone from "" to "November 2016"
* jgonzalez commented:

> ok... this is indeed a reasonable request for I3CorsikaInjector. I am assigning a milestone around November so we can discuss the whole CORSIKA thing at the meeting/simulation workshop. What might end up happening is that we introduce the new reader and modify I3CorsikaInjector to work with it, and in this case the thing will be solved.


* jgonzalez changed priority from "critical" to "normal"

**Comment by icecube-trac** [comment]
Trac update at 20170331T15:00:28: jgonzalez changed milestone from "November 2016" to "IceSim 6"

**Comment by icecube-trac** [comment]
Trac update at 20180706T18:15:07: olivas changed milestone from "IceSim 6" to "Autumnal Equinox 2018"

**Comment by icecube-trac** [comment]
Trac update at 20181001T21:19:09: 
* olivas commented:

> The future of this ticket is uncertain.  We'll discuss on the next phone call.


* olivas changed summary from "IceTop detector simulation: I3CorsikaInjector and .bz2 files" to "[IceTop] IceTop detector simulation: I3CorsikaInjector and .bz2 files"

**Comment by icecube-trac** [comment]
Trac update at 20181010T20:18:09: 
* olivas changed milestone from "Autumnal Equinox 2018" to "Winter Solstice 2018"
* olivas commented:

> Note this have moved to: https://github.com/IceCubeOpenSource/corsika_reader
> 
> Pushing this to the next milestone.

**Comment by icecube-trac** [comment]
Trac update at 20190320T22:11:27: 
* olivas changed status from "assigned" to "closed"
* olivas changed resolution from "" to "duplicate"
* olivas commented:

> This ticket has been resubmitted as a GitHub issue.
> https://github.com/IceCubeOpenSource/corsika_reader/issues/17

**Comment by icecube-trac** [comment]
Trac update at 20190321T00:01:54: jgonzalez commented:

> Not exactly :)
> 
> The code in github already knows how to handle bz2 files. What does not handle it is the I3CorsikaInjector, which lives in topsimulator. I3CorsikaInjector does not handle it because it does not use the reader in github but an older reader that lives in topsimulator.

**Comment by icecube-trac** [comment]
Trac update at 20190321T01:17:28: 
* olivas changed status from "closed" to "reopened"
* olivas commented:

> Ugh.


* olivas changed resolution from "duplicate" to ""

**Comment by icecube-trac** [comment]
Trac update at 20190321T01:38:32: 
* olivas commented:

> topsimulator needs a maintainer, which isn't going to happen before Friday.  Pushing this to the next milestone.


* olivas changed summary from "[IceTop] IceTop detector simulation: I3CorsikaInjector and .bz2 files" to "[topsimulator] IceTop detector simulation: I3CorsikaInjector and .bz2 files"
* olivas changed milestone from "Vernal Equinox 2019" to "Summer Solstice 2019"

**Comment by icecube-trac** [comment]
Trac update at 20190321T16:04:40: 
* olivas changed owner from "jgonzalez" to "mplum"
* olivas changed status from "reopened" to "assigned"

**Comment by icecube-trac** [comment]
Trac update at 20190604T18:49:20: olivas commented:

> Owner: Please 'accept' this ticket acknowledging you're aware of its existence.

**Comment by icecube-trac** [comment]
Trac update at 20190605T02:22:02: mplum changed status from "assigned" to "accepted"

**Comment by icecube-trac** [comment]
Trac update at 20191216T16:06:24: 
* mplum changed milestone from "Winter Solstice 2019" to "Vernal Equinox 2020"
* mplum changed priority from "normal" to "minor"

**Comment by icecube-trac** [comment]
Trac update at 20200624T12:31:42: 
* olivas commented:

> Milestone Vernal Equinox 2020 deleted


* olivas changed milestone from "Vernal Equinox 2020" to "Autumnal Equinox 2020"
